homeassistant/components/media_player/sonos.py

```diff
@@ -138,9 +138,14 @@ def update(self):
         """Retrieve latest state."""
         self._name = self._player.get_speaker_info()['zone_name'].replace(
             ' (R)', '').replace(' (L)', '')
-        self._status = self._player.get_current_transport_info().get(
-            'current_transport_state')
-        self._trackinfo = self._player.get_current_track_info()
+
+        if self._is_reachable:
+            self._status = self._player.get_current_transport_info().get(
```

Comment:
> Wouldn't this call raise an error if we can't connect ? So why not just wrap this block in a try … except ?

As I stated in the commit message (and PR description) this code can raise 3 different type of exceptions at different points in time. The 2nd exception can be raised while we are handling the 1st one, we would have to write nested `try...except` blocks :(